generate/deps.go

```diff
@@ -132,7 +132,18 @@ func (u *Update) localDep(conf *config.Config, importPath string) (string, error
 			continue
 		}
 
-		if kind.Type == kinds.Lib {
+		if kind.Type != kinds.Lib {
+			continue
+		}
+
+		name := rule.Name()
+		pkg := rule.Attr("package")
```

Comment:
Ah sorry, I think you've fallen afoul of some legacy code here. The `package` argument doesn't actually do anything meaningful unless you have `LegacyImports` enabled. If you want to figure out the package for a library target, you have to parse the `package foo` statement out of the `.go` files, like we do in `(*Update) rulePkg()`. 

For context, we used to compile packages differently to `go build` where we'd add another path segment to the import path as needed, such that the base name of the import path always matched the package name. This is no longer the case. Now the import path is based solely off the directory path, just like it is for `go build`.

Reply:
Oh great! I didn't notice this change.